armi/utils/plotting.py

```diff
@@ -1047,3 +1063,304 @@ def makeHistogram(x, y):
+                patch = matplotlib.patches.Polygon(corners, fill=True)
+                blockPatch.append(patch)
+        else:
+            # Just make it a hexagon...
+            blockPatch = matplotlib.patches.RegularPolygon(
+                (x, y), 6, component.getDimension("op") / math.sqrt(3)
+            )
+
+    elif isinstance(component, Rectangle):
+        if component.getDimension("widthInner") != 0:
+            innerPoints = numpy.array(
+                [
+                    [
+                        x + component.getDimension("widthInner") / 2,
+                        y + component.getDimension("lengthInner") / 2,
+                    ],
+                    [
+                        x + component.getDimension("widthInner") / 2,
+                        y - component.getDimension("lengthInner") / 2,
+                    ],
+                    [
+                        x - component.getDimension("widthInner") / 2,
+                        y - component.getDimension("lengthInner") / 2,
+                    ],
+                    [
+                        x - component.getDimension("widthInner") / 2,
+                        y + component.getDimension("lengthInner") / 2,
+                    ],
+                ]
+            )
+
+            outerPoints = numpy.array(
+                [
+                    [
+                        x + component.getDimension("widthOuter") / 2,
+                        y + component.getDimension("lengthOuter") / 2,
+                    ],
+                    [
+                        x + component.getDimension("widthOuter") / 2,
+                        y - component.getDimension("lengthOuter") / 2,
+                    ],
+                    [
+                        x - component.getDimension("widthOuter") / 2,
+                        y - component.getDimension("lengthOuter") / 2,
+                    ],
+                    [
+                        x - component.getDimension("widthOuter") / 2,
+                        y + component.getDimension("lengthOuter") / 2,
+                    ],
+                ]
+            )
+            blockPatch = []
+            for n in range(4):
+                corners = [
+                    innerPoints[n],
+                    innerPoints[(n + 1) % 4],
+                    outerPoints[(n + 1) % 4],
+                    outerPoints[n],
+                ]
+                patch = matplotlib.patches.Polygon(corners, fill=True)
+                blockPatch.append(patch)
+        else:
+            # Just make it a rectangle...
+            blockPatch = matplotlib.patches.Rectangle(
+                (
+                    x - component.getDimension("widthOuter", cold) / 2,
+                    y - component.getDimension("lengthOuter", cold) / 2,
+                ),
+                component.getDimension("widthOuter", cold),
+                component.getDimension("lengthOuter", cold),
+            )
+    if type(blockPatch) is not list:
```

Comment:
pylint is probably complaining about this. Prefer `if isinstance(blockPatch, list):` when possible. The direct type comparison is very similar, but does not test true for sub-classes of `list`. Sometimes, but rarely, this is necessary, so it looks a bit strange to see if that logic isn't needed.